### DISCUSSION TITLE
Filter allowed blocks when page template is selected

### DISCUSSION THREAD

**Discussion by souricardomaia** [WordPress/gutenberg#37481] [Developer Experience]
![Captura de Tela 2021-12-17 às 06 59 52](https://user-images.githubusercontent.com/61801718/146527636-dce06b9f-fd8f-4687-81e8-b33372d52fa4.png)

Hey there,
how i can filter allowed blocks when a specific page template is selected?

It's possible?

**Comment by t-hamano** [comment]
Hi @rmaaia,

I have not tried that, but it may be possible by combining the following two approaches:

- Get the current template slug with [getEditedPostAttribute('template')
](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/data/data-core-editor.md#geteditedpostattribute)
- Use [blockEditor.__unstableCanInsertBlockType](https://github.com/WordPress/gutenberg/blob/204cd8b9411171e703f1c751031b6c243e4d5966/packages/block-editor/src/store/selectors.js#L1548-L1575) filter to control blocks that can be inserted
